docs/development/decision-records/2024-08-28-Release-Workflow-Refactor/README.md

```diff
@@ -0,0 +1,82 @@
+to keep the integrity of the commit tree. Finally, the current project version is automatically bumped to the next minor version in a new commit to `main` 
+to kickstart the next development cycle.
+
+As they stand, the current release workflows don't allow the release of bugfix versions since:
+- the `draft-new-release` workflow doesn't create any bugfix branch;
+- even if the branch is manually created, merging it to `releases` via PR to trigger the `publish-new-release` workflow, is not possible;
+- The `publish-new-release` workflow can't be manually triggered;
+- Even if it could be manually triggered, there are steps that shouldn't apply for bugfix releases such as bumping the project version or merging to main;
+
+
+## Approach
+
+The draft-new-release workflow should include one extra inputs:
+- A `choice` input to allow the distinction between a release or bugfix.
+
+Github Actions Jobs and Step conditionals are then used to make the execution follow the required steps, as follows.
+
+### Prepare Release workflow logic
+- Triggers manually. Input is the version, choice and base ref (should be either commit from main or tag)
+- if choice is release
+  - branches from main into `release/<input_version>`
+- if choice is bugfix
+  - branches from tag into `bugfix/<input_version>`
+- replaces the project default version `(gradle.properties)` with the workflow version input
+- replaces openapi spec version
+- replaces charts version and README
+- commits and pushes to `release/*` or `bugfix/X.Y.Z` branch
+- if new release
+   - opens PR with new `release/*` branch as head and `main/` as base
+
+Then, a different process takes places if we intend to release a latest version or a bugfix. 
+
+### Case for latest releases
+After the prepare release PR is approved, merging into main will trigger the publish-release workflow where the required publication steps will take place, as shown in [Publish release workflow logic](#publish-release-workflow-logic).
+
+### Case for bugfix Releases
+After the bugfix branch is created, developers commit the fix or set of fixes to it via PR.
+The release of the bugfix version should be triggered manually through the publishing workflow and should execute most of the publication steps as a normal release.
+However, for a bugfix release, the workflow shouldn't:
+- Make any commits to the main branch
+- Bump the project version to the next minor version
+
+### Publish release workflow logic
+- Triggers automatically when prepare release PR is merged into main from `release/*` branch
+- Triggers manually only from branch `bugfix/X.Y.Z`
+
+- Extracts release version from `gradle.properties`
+- Triggers release to maven repository by calling `trigger-maven-publish.yaml`
+- Triggers docker image publishing by calling `trigger-maven-publish.yaml`
+- Releases helm-charts
+- Creates Github Tag and Release
+- if PR was merged to main it means a new release then:
```

Comment:
I think that it's better to create a new workflow `publish-new-hotfix`, that only triggers manually and lacks the steps to bump versions, etc.

My reasoning is that I find it a bit sketchy to parse the head ref in the workflow, and only perform these tasks when the head ref starts with `release`

Reply:
This would mean to have duplicate workflows with minor changes between them. I would prefer to parse the head ref instead as its simple to.

Reply:
to avoid code duplication, commonalities could be extracted into a re-usable workflow, which then gets parameterized. but having explicit workflows for release and bugfix has advantages:
- less error prone
- possibility to assign individual triggers and guards (e.g. disallowing releases on refs other than `main`)
- individual WF names: useful if another WF should be triggered after this WF (`workflow_run` trigger)

Reply:
Ok, i see.
Let me know what you think of these changes: [8013954](https://github.com/eclipse-tractusx/tractusx-edc/pull/1521/commits/801395426b870ac9c902ca16400b14d26a2b9950)